### PR TITLE
signal from setState function

### PR DESCRIPTION
**Pull request by dorezyuk**
I've moved the condition-variable inside setState, so we can easier trigger it on state-update. This simplifies the code a bit

### CODE REVIEW

**Review by corot — CHANGES_REQUESTED** [review]
For controller, I had a different idea in mind: create a setFailedState method that joins these three lines:
```
          setState(<any error state>);
          publishZeroVelocity();
          condition_.notify_all();
          moving_ = false;
```
Does it make sense the same for planner?
In any case, whatever refactor we do, should be done also in recovery and controlling executions

**Comment by dorezyuk** [comment]
I wouldn't mix the stuff and do the refactoring step by step. Right now there is no correspondent action to publishZeroVelocity for planners failing. We might later refactor the code into
```
setState(State _state, bool _signalling) {
...
planning_ = !_signalling
}
```
since we typically want to exit the planning loop if we signal

**Comment by corot** [comment]
> I wouldn't mix the stuff and do the refactoring step by step. Right now there is no correspondent action to publishZeroVelocity for planners failing. We might later refactor the code into

As long as you follow up this PR with the matching changes for controller and recovery, I'm good approving this

**Review by corot — APPROVED** [review]

File: mbf_abstract_nav/src/abstract_planner_execution.cpp [hunk 1]
```diff
@@ -106,10 +106,14 @@ typename AbstractPlannerExecution::PlanningState AbstractPlannerExecution::getSt
...
 }
 
-void AbstractPlannerExecution::setState(PlanningState state)
+void AbstractPlannerExecution::setState(PlanningState state, bool signalling)
 {
   boost::lock_guard<boost::mutex> guard(state_mtx_);
   state_ = state;
+
+  // some states are quiet, most aren't
+  if(signalling)
```
Comment (corot) [inline mbf_abstract_nav/src/abstract_planner_execution.cpp:115]: missing space after if

Comment (dorezyuk) [reply mbf_abstract_nav/src/abstract_planner_execution.cpp:115]: would it make sense to add a clang-format file so we use common formatting?

Comment (corot) [reply mbf_abstract_nav/src/abstract_planner_execution.cpp:115]: yes,,,  #239

**Review by spuetz — APPROVED** [review]

**Pull request merged**